mypy/checker.py

```diff
@@ -3577,9 +3578,14 @@ def check_subtype(self, subtype: Type, supertype: Type, context: Context,
                     extra_info.append(supertype_label + ' ' + supertype_str)
                 note_msg = make_inferred_type_note(context, subtype,
                                                    supertype, supertype_str)
+                if msg == message_registry.INCOMPATIBLE_TYPES_IN_ASSIGNMENT:
+                    notes = append_invariance_notes([], subtype,  supertype)
             if extra_info:
                 msg += ' (' + ', '.join(extra_info) + ')'
             self.fail(msg, context)
+            if notes:
```

Comment:
This `if` is redundant. If the list is empty, the loop body will simply not run.

Reply:
Good I refactored it but I didn't improve this part. Thank you.